tests/test_mariadb.py

```diff
@@ -276,3 +280,108 @@ def test_mariadb_healthcheck_galera_cluster_disabled(auto_container_per_test):
     _wait_for_server(conn)
 
     conn.run_expect([1], "healthcheck.sh --su-mysql --galera_online")
+
+
+_DB_ENV = {
+    "MARIADB_USER": _OTHER_DB_USER,
+    "MARIADB_PASSWORD": _OTHER_DB_PW,
+    "MARIADB_DATABASE": _TEST_DB,
+    "MARIADB_ROOT_PASSWORD": MARIADB_ROOT_PASSWORD,
+    "MARIADB_AUTO_UPGRADE": "1",
+}
+
+
+@pytest.mark.parametrize("ctr_image", MARIADB_CONTAINERS)
+@pytest.mark.skipif(
+    OS_VERSION in ("15.5",), reason="MariaDB upgrade scenario not supported"
+)
+def test_mariadb_upgrade(
+    container_runtime: OciRuntimeBase,
+    pytestconfig: pytest.Config,
+    ctr_image: DerivedContainer,
+    tmp_path: Path,
+) -> None:
+    mounts = [BindMount(host_path=tmp_path, container_path="/var/lib/mysql")]
+    mariadb_old = DerivedContainer(
+        base="registry.suse.com/suse/mariadb:10.6",
+        containerfile='RUN set -euo pipefail; head -n -1 /usr/local/bin/gosu > /tmp/gosu;  echo \'exec setpriv --pdeathsig=keep --reuid="$u" --regid="$u" --clear-groups -- "$@"\' >> /tmp/gosu;  mv /tmp/gosu /usr/local/bin/gosu; chmod +x /usr/local/bin/gosu',
+        volume_mounts=mounts,
+        extra_environment_variables=_DB_ENV,
+    )
+    mariadb_new = DerivedContainer(
+        base=ctr_image,
+        volume_mounts=mounts,
+        extra_environment_variables=_DB_ENV,
+    )
+    with ContainerLauncher.from_pytestconfig(
+        mariadb_old, container_runtime, pytestconfig
+    ) as launcher:
+        launcher.launch_container()
+        con = launcher.container_data.connection
+        _wait_for_server(con)
+        mariadb_cmd = f"mariadb --user={_OTHER_DB_USER} --password={_OTHER_DB_PW} --host=0.0.0.0 {_TEST_DB}"
+
+        con.check_output(
+            f'echo "CREATE TABLE random_strings (string VARCHAR(255) NOT NULL);" | {mariadb_cmd}'
+        )
+
+        for _ in range(10):
+            con.check_output(
+            f"echo 'INSERT INTO random_strings (string) VALUES (MD5(RAND()));' | {mariadb_cmd}"
+            )
+
+        rows = (
+            con.check_output(f'echo "SELECT count(*) FROM random_strings;" | {mariadb_cmd}')
+            .strip()
+            .splitlines()
+        )
+
+        assert rows and len(rows) == 2
+        _, count = rows[-1].split()
+        assert count == "10"
```

Comment:
`SELECT count(*) FROM random_strings;` returns the following:
```
count(*)
10
```

i.e. the `strip().splitlines()` yields: ["count(*)", "10"], hence the assert on `rows and len(rows) == 2` passes, but to obtain count just run: `count = rows[-1]`.

This also needs to be changed in the four places below (you might want to put this into a separate function)

Reply:
fixed.